npc_gzip/compressors/base.py

```diff
@@ -0,0 +1,115 @@
+import os
+from types import ModuleType
+
+from npc_gzip.exceptions import InvalidCompressorException
+
+
+class BaseCompressor:
+    """
+    Default compressor class that other compressors inherit
+    from.
+
+    >>> import gzip
+
+    >>> compressor = BaseCompressor(compressor=gzip)
+
+    >>> example = "Hello there!"
+    >>> compressed_length: int = compressor.get_compressed_length(example)
+    >>> bits_per_character: float = compressor.get_bits_per_character(example)
+    >>> assert isinstance(compressed_length, int)
+    >>> assert isinstance(bits_per_character, float)
+    """
+
+    def __init__(self, compressor: ModuleType) -> None:
+        if not isinstance(compressor, (ModuleType, BaseCompressor)):
+            raise InvalidCompressorException(
+                f"Not a function passed: {type(compressor)}"
+            )
+        self.compressor = compressor
+
+    def _open_file(self, filepath: str, as_bytes: bool = False):
+        """
+        Helper function that loads and returns the contents
+        of a file at `filepath`. Optional `as_bytes` parameter
+        will load the file contents with `open(filepath, 'rb')`
+        if True.
+
+        Arguments:
+            filepath (str): Path to the file to read contents from.
+            as_bytes (bool): [Optional] If true, opens the file as bytes.
+
+        Returns:
+            str: File contents as a string.
+        """
+
+        assert os.path.isfile(filepath), f"Filepath ({filepath}) does not exist."
+        file_contents = None
+        open_mode = "r"
+        if as_bytes:
+            open_mode = "rb"
+
+        with open(filepath, open_mode) as f:
+            file_contents = f.read()
+            f.close()
+
+        if as_bytes:
+            file_contents = file_contents.decode("utf-8")
+
+        return file_contents
+
+    def _compress(self, x: str) -> bytes:
+        """
+        Applies the compression algorithm to `x` and
+        returns the results as bytes.
+
+        Arguments:
+            x (str): The string you want to compress.
+
+        Returns:
+            bytes: The compressed bytes representation of `x`.
+        """
+
+        x = str(x)
```

Comment:
As with some of the other type conversions, I think it may be best not to have this. See my comment on the corresponding `x = str(x)` statement in `get_bits_per_character` later in this file for details.